pkg/syncclient/sync_client.go

```diff
@@ -41,17 +70,24 @@ func New(addr string, myVersion, myHostname, myInfo string, cbs api.SyncerCallba
 		myVersion:  myVersion,
 		myHostname: myHostname,
 		myInfo:     myInfo,
+
+		options: options,
```

Comment:
Just checking my understanding here - we don't need any defaulting of this field because the functions operate on a `*Options` even if that pointer is `nil` and we've made sure that the functions handle the `o == nil` case.

I think this is fine, but is there a reason we like this better than just using a default `Options` here?

for example something like this? 

```go
if options == nil {
  options = &Options{defaultReadTimeout, defaultWriteTimeout}
}
```

Reply:
Correct.  I've tried defaulting config in various different ways; I haven't really found an approach that seems like a clear winner.

On this occasion, I was following the Gopher philosophy tenet "the nil value should be useful".